docs/data/date-pickers/validation/DateValidationShouldDisableMonth.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+import Box from '@mui/material/Box';
+import Stack from '@mui/material/Stack';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
+
+const today = dayjs();
+
+const isInCurrentMonth = (date: Dayjs) => date.get('month') === dayjs().get('month');
+
+const GridItem = ({
+  label,
+  children,
+  spacing = 1,
+}: {
+  label: string;
+  children: React.ReactNode;
+  spacing?: number;
+}) => {
+  return (
+    <Box sx={{ display: 'flex', flexDirection: 'column' }}>
+      <Typography variant="body2" sx={{ mb: spacing }}>
+        {label}
+      </Typography>
+      {children}
+    </Box>
+  );
+};
+
+export default function DateValidationShouldDisableMonth() {
+  const [datePickerValue, setDatePickerValue] = React.useState<Dayjs | null>(today);
+  const [dateTimePickerValue, setDateTimePickerValue] = React.useState<Dayjs | null>(
+    today,
+  );
+
+  return (
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <Stack spacing={4}>
+        <GridItem label="DatePicker">
+          <DatePicker
+            shouldDisableDate={isInCurrentMonth}
```

Comment:
```suggestion
            shouldDisableMonth={isInCurrentMonth}
```

Same on Year demo. It seems that `DateTimePicker` does not support the `shouldDisableYear`

Reply:
Indeed, I passed the wrong prop :grimacing: 

> It seems that DateTimePicker does not support the shouldDisableYear

What makes you think that
I just tried it and for me it works

Reply:
The days are disabled, but the TextField is not in error mode

![image](https://user-images.githubusercontent.com/45398769/191947824-e3a5d463-de38-426b-96a5-fd27be758aae.png)

Reply:
Oooohhhh

Reply:
https://github.com/mui/mui-x/blob/44eb781b15bb3c2c04c4413830f42fc7df27b7f9/packages/x-date-pickers/src/internals/hooks/validation/useDateTimeValidation.ts#L32

Do you see a reason not to pass every prop to both date and time validation ?
That's the cause of the bug.

By the way, even if this page was never published, we found lots of bugs writing and reviewing it :laughing:

Reply:
Sounds good to pass everything and let hooks peak what they need.

Reply:
> On the same topic, I was surprised that DatePickers do not support maxDateTime/minDateTime

Right now `minDateTime` / `maxDateTime` management is quite ugly, it's just replaced in `DateTimePicker` which mean even it can't use both `minDateTime` and `minDate` at the same time for instance.

For `DatePicker` what would be the behavior for you ?
Would it validate the time when passed `maxDateTime` / `minDateTime` or just the date ?

Reply:
Intuitively I would have expected it to validate the date. But it does not make a lot of sense since it would have the same behavior as `maxDate`